src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,342 @@
+'use client';
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { getDaysInMonth, isToday, isSameDay } from 'date-fns';
+
+type ClientScheduleAppointmentProps = {
+    availableDates: Date[];
+};
+
+const MONTH_NAMES = [
+    'Janeiro',
+    'Fevereiro',
+    'Março',
+    'Abril',
+    'Maio',
+    'Junho',
+    'Julho',
+    'Agosto',
+    'Setembro',
+    'Outubro',
+    'Novembro',
+    'Dezembro',
+];
+
+const DAYS = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];
+
+// Calculates the number of days in the given month
+const getNoOfDays = (date: Date) => {
+    const daysInMonth = getDaysInMonth(date);
+    const dayOfWeek = date.getDay();
+
+    const blankdaysArray: number[] = [];
+    for (var i = 1; i <= dayOfWeek; i++) {
+        blankdaysArray.push(i);
+    }
+
+    const daysArray: number[] = [];
+    for (var i = 1; i <= daysInMonth; i++) {
+        daysArray.push(i);
+    }
+
+    return { blankDays: blankdaysArray, numDays: daysArray };
+};
+
+const isSelectedDate = (newDate: Date, selectedDate: Date) => {
+    return isSameDay(newDate, selectedDate);
+};
+
+const isSelectedHour = (hour: number, selectedHour: number | undefined) => {
+    return selectedHour === hour;
+};
+
+const submitDate = (selectedDate: Date) => {
+    if (selectedDate) {
+        //send the selectedDate object
+    }
+};
```

Comment:
submitDate should get more than the date. 

We will continue the flow by attaching the user to an availability, this is more than just the date, we need the id...